### ISSUE TITLE
Warnings during libgit2 tests

### ISSUE THREAD

**Issue by quinnj** [JuliaLang/julia#22702]
```julia
    JULIA test/libgit2
Test (Worker) | Time (s) | GC (s) | GC % | Alloc (MB) | RSS (MB)
WARNING: readuntil(IO,AbstractString) will perform poorly with a long string
WARNING: readuntil(IO,AbstractString) will perform poorly with a long string
WARNING: readuntil(IO,AbstractString) will perform poorly with a long string
libgit2 (1)   |   30.68  |  0.28  |  0.9 | 443.88     | 473.48

Test Summary: | Pass  Total
  Overall     |  449    449
    SUCCESS```

**Comment by tkelman** [comment]
https://github.com/JuliaLang/julia/pull/20738#issuecomment-293033088, #20621

**Comment by omus** [comment]
I need to get #20621 wrapped up...

**Comment by omus** [comment]
Fixed by #20621
